docs/topics/test-system.md

```diff
@@ -1,72 +1,126 @@
-WDQS_PROXY_IMAGE_NAME=ghcr.io/wmde/wdqs-proxy:1157808966
-MYSQL_IMAGE_NAME=mariadb:10.3
-```
-
-### 3. start containers again 
+Do the following (with the parameters you require)...
 
 ```sh
-docker-compose -f docker-compose.yml -f docker-compose.extra.yml up -d
+# Inputs for setup
+TEST_SYSTEM=latest
+EXAMPLE_HASH=b8aa96cb0cd99054631b558535ef1f3a9b8d41b8
+BUILD_NUMBER=1824280943
+
+TEST_SYSTEM=previous
+EXAMPLE_HASH=b8aa96cb0cd99054631b558535ef1f3a9b8d41b8
+BUILD_NUMBER=1853048237
+
+# Calculate some things
+PORT_BASE="83"
+if [ "$TEST_SYSTEM" == "previous" ];
+then
+    # Only suffix domains of the "previous" system
+    DOMAIN_SUFFIX=-$TEST_SYSTEM
+    PORT_BASE="82"
+fi
+
+umask 002
+mkdir -p /opt/test-systems/$TEST_SYSTEM
+cd /opt/test-systems/$TEST_SYSTEM
+
+# Download the repo at the desired version, and extract the example
+wget https://github.com/wmde/wikibase-release-pipeline/archive/$EXAMPLE_HASH.zip
+unzip $EXAMPLE_HASH.zip
+rm $EXAMPLE_HASH.zip
+cp -r ./wikibase-release-pipeline-$EXAMPLE_HASH/example/* .
+rm -rf ./wikibase-release-pipeline-$EXAMPLE_HASH
+
+# Create a .env file
+cp ./template.env ./.env
+echo "# Test system customizations" >> ./.env
+# Default settings to change
+echo "MW_WG_ENABLE_UPLOADS=true" >> ./.env
+# Public facing domains
+echo "WIKIBASE_HOST=wikibase-product-testing$DOMAIN_SUFFIX.wmflabs.org" >> ./.env
+echo "WDQS_FRONTEND_HOST=wikibase-query-testing$DOMAIN_SUFFIX.wmflabs.org" >> ./.env
+echo "QUICKSTATEMENTS_HOST=wikibase-qs-testing$DOMAIN_SUFFIX.wmflabs.org" >> ./.env
+# Images to use
+echo "WIKIBASE_IMAGE_NAME=ghcr.io/wmde/wikibase:$BUILD_NUMBER" >> ./.env
+echo "WDQS_IMAGE_NAME=ghcr.io/wmde/wdqs:$BUILD_NUMBER" >> ./.env
+echo "WDQS_FRONTEND_IMAGE_NAME=ghcr.io/wmde/wdqs-frontend:$BUILD_NUMBER" >> ./.env
+echo "ELASTICSEARCH_IMAGE_NAME=ghcr.io/wmde/elasticsearch:$BUILD_NUMBER" >> ./.env
+echo "WIKIBASE_BUNDLE_IMAGE_NAME=ghcr.io/wmde/wikibase-bundle:$BUILD_NUMBER" >> ./.env
+echo "QUICKSTATEMENTS_IMAGE_NAME=ghcr.io/wmde/quickstatements:$BUILD_NUMBER" >> ./.env
+echo "WDQS_PROXY_IMAGE_NAME=ghcr.io/wmde/wdqs-proxy:$BUILD_NUMBER" >> ./.env
+# Ports to expose
+echo "WIKIBASE_PORT=${PORT_BASE}80" >> ./.env
+echo "WDQS_FRONTEND_PORT=${PORT_BASE}81" >> ./.env
+echo "QS_PUBLIC_SCHEME_HOST_AND_PORT=https://wikibase-qs-testing$DOMAIN_SUFFIX.wmflabs.org:${PORT_BASE}82" >> ./.env
+echo "QUICKSTATEMENTS_PORT=${PORT_BASE}82" >> ./.env
+
+# Create an extra LocalSettings.php file to load
+wget https://gist.githubusercontent.com/addshore/760b770427eb81d4d1ee14eb331246ea/raw/92a464ca0ee6b2432cc1af9c977166f794c78488/gistfile1.txt
+mv gistfile1.txt ./extra.LocalSettings.php
+sed -i 's/#- .\/LocalSettings.php:\/var\/www\/html\/LocalSettings.d\/LocalSettings.override.php/- .\/extra.LocalSettings.php:\/var\/www\/html\/LocalSettings.d\/LocalSettings.extra.php/' ./docker-compose.yml
 ```
 
-### 4. Check logs to see what happens
+You should the be able to start the test system from your regular user.
```

Comment:
But in the example we start it with sudo

Reply:
This was written as before the other steps were intended to be run as the `mediawiki` user.
I changed this now so will rewrite this sentence!